app/api/orders.py

```diff
@@ -319,7 +319,7 @@ def before_update_object(self, order, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        if data.get('amount') or data.get('is_billing_enabled'):
+        if data.get('amount') or data.get('is_billing_enabled') or order.event.is_billing_info_mandatory:
```

Comment:
There's a problem, this will require billing info on any paid order, whereas it is explicitly stated in the issue to make it optional unless is_billing_info_mandatory is True.

Skipping this PR for now, will add in minor version

Reply:
Merging it after release would be better I guess too.